x-pack/test/cloud_security_posture_api/routes/csp_benchmark_rules_bulk_update.ts

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { expect as expectExpect } from 'expect';
+
+import {
+  ELASTIC_HTTP_VERSION_HEADER,
+  X_ELASTIC_INTERNAL_ORIGIN_REQUEST,
+} from '@kbn/core-http-common';
+import type { FtrProviderContext } from '../ftr_provider_context';
+
+interface RuleIdentifier {
+  benchmarkId: string;
+  benchmarkVersion: string;
+  ruleNumber: string;
+}
+
+// eslint-disable-next-line import/no-default-export
+export default function ({ getService }: FtrProviderContext) {
+  const retry = getService('retry');
+  const supertest = getService('supertest');
+  const log = getService('log');
+
+  const generateRuleKey = (ruleParams: RuleIdentifier): string => {
+    return `${ruleParams.benchmarkId};${ruleParams.benchmarkVersion};${ruleParams.ruleNumber}`;
+  };
+
+  const generateRandomRule = (): RuleIdentifier => {
+    const majorVersionNumber = Math.floor(Math.random() * 10); // Random major number between 0 and 9
+    const minorVersionNumber = Math.floor(Math.random() * 10);
+    const benchmarksIds = ['cis_aws', 'cis_k8s', 'cis_k8s'];
+    const benchmarksVersions = ['v2.0.0', 'v2.0.1', 'v2.0.3', 'v3.0.0'];
+    const randomBenchmarkId = benchmarksIds[Math.floor(Math.random() * benchmarksIds.length)];
+    const randomBenchmarkVersion =
+      benchmarksVersions[Math.floor(Math.random() * benchmarksVersions.length)];
+
+    return {
+      benchmarkId: randomBenchmarkId,
+      benchmarkVersion: randomBenchmarkVersion,
+      ruleNumber: `${majorVersionNumber}.${minorVersionNumber}`,
+    };
+  };
+
+  /**
+   * required before indexing findings
+   */
+  const waitForPluginInitialized = (): Promise<void> =>
+    retry.try(async () => {
+      log.debug('Check CSP plugin is initialized');
+      const response = await supertest
+        .get('/internal/cloud_security_posture/status?check=init')
+        .set(ELASTIC_HTTP_VERSION_HEADER, '1')
+        .expect(200);
+      expect(response.body).to.eql({ isPluginInitialized: true });
+      log.debug('CSP plugin is initialized');
+    });
+
+  describe('Verify update csp rules states API', async () => {
+    before(async () => {
+      await waitForPluginInitialized();
+    });
+
+    afterEach(async () => {});
```

Comment:
missing cleanup after each test

if you'll print the settings you'll see you have rule states from previous test

Reply:
The only way to do it is via the saved object API, and It's not possible since the saved object API is not available anymore.

Reply:
I see that SO API is exposed through `kibanaServer` service on other FTR tests and is being used to clear the data between tests

In your use case it should look something like:

```ts
      await kibanaServer.savedObjects.clean({
        types: ['cloud-security-posture-settings']
      });
```